### PR TITLE
Implemented Official Summer 2 Costume

### PR DESCRIPTION
**Pull request by Jedzkie**
[//]: # (**********************************)
[//]: # (** Fill in the following fields **)
[//]: # (**********************************)

[//]: # (Note: Lines beginning with syntax such as this one, are comments and will not be visible in your report!)

### Pull Request Prelude

[//]: # (Thank you for working on improving Hercules!)

[//]: # (Please complete these steps and check the following boxes by putting an `x` inside the brackets _before_ filing your Pull Request.)

- [x] I have followed [proper Hercules code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR will be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed

[//]: # (Describe at length, the changes that this pull request makes.)

**Affected Branches:** master

[//]: # (Master? Slave?)

**Issues addressed:**
https://github.com/HerculesWS/Hercules/issues/1683
http://herc.ws/board/topic/6818-summer-2-costume-job/

[//]: # (Issue Tracker Number if any.)

### Known Issues and TODO List
- [x] Ignore palette configuration for Oktoberfest
- [x] Coding-style Fixes and White space adjustments

[//]: # (Insert checklist here)
[//]: # (Syntax: - [ ] Checkbox)

[//]: # (**NOTE** Enable the setting "[√] Allow edits from maintainers." when creating your pull request if you have not already enabled it.)

[cont]: https://github.com/HerculesWS/Hercules/blob/master/CONTRIBUTING.md
[code]: https://github.com/HerculesWS/Hercules/wiki/Coding-Style

### CODE REVIEW

**Comment by HerculesWSAPI** [comment]
This change is [Reviewable](https://reviewable.io/reviews/herculesws/hercules/1699)

File: src/common/mmo.h [hunk 1]
```diff
@@ -447,7 +448,7 @@ enum {
 #endif
 	// compound constants
 	OPTION_DRAGON    = OPTION_DRAGON1|OPTION_DRAGON2|OPTION_DRAGON3|OPTION_DRAGON4|OPTION_DRAGON5,
-	OPTION_COSTUME   = OPTION_WEDDING|OPTION_XMAS|OPTION_SUMMER|OPTION_HANBOK|OPTION_OKTOBERFEST,
+	OPTION_COSTUME   = OPTION_WEDDING|OPTION_XMAS|OPTION_SUMMER|OPTION_HANBOK|OPTION_OKTOBERFEST|OPTION_SUMMER2,
```
Comment (Smokexyz) [inline src/common/mmo.h]: Spaces between binary bitwise operators.

File: src/map/clif.c [hunk 2]
```diff
@@ -3215,10 +3215,12 @@ void clif_changelook(struct block_list *bl,int type,int val)
 					vd->cloth_color = 0;
 				if (sd->sc.option&OPTION_OKTOBERFEST /* TODO: config? */)
 					vd->cloth_color = 0;
+				if (sd->sc.option&OPTION_SUMMER2 && battle_config.summer2_ignorepalette)
```
Comment (Smokexyz) [inline src/map/clif.c]: Spaces between binary bitwise operators.

File: src/map/clif.c [hunk 3]
```diff
@@ -3215,10 +3215,12 @@ void clif_changelook(struct block_list *bl,int type,int val)
...
+					vd->cloth_color = 0;
 				if (vd->body_style && (
 					sd->sc.option&OPTION_WEDDING || sd->sc.option&OPTION_XMAS ||
 					sd->sc.option&OPTION_SUMMER || sd->sc.option&OPTION_HANBOK ||
-					sd->sc.option&OPTION_OKTOBERFEST))
+					sd->sc.option&OPTION_OKTOBERFEST || sd->sc.option&OPTION_SUMMER2))
```
Comment (Smokexyz) [inline src/map/clif.c]: Spaces between binary bitwise operators.

File: src/map/status.c [hunk 4]
```diff
@@ -6883,13 +6885,16 @@ void status_set_viewdata(struct block_list *bl, int class_)
 				       sd->vd.cloth_color = 0;
 				if (sd->sc.option&OPTION_OKTOBERFEST /* TODO: config? */)
 					sd->vd.cloth_color = 0;
+				if (sd->sc.option&OPTION_SUMMER2 && battle_config.summer2_ignorepalette)
```
Comment (Smokexyz) [inline src/map/status.c]: Spaces between binary bitwise operators.

File: src/map/status.c [hunk 5]
```diff
@@ -6883,13 +6885,16 @@ void status_set_viewdata(struct block_list *bl, int class_)
...
+					sd->vd.cloth_color = 0;
 			}
 			if (sd->vd.body_style
 			 && (sd->sc.option&OPTION_WEDDING
 			  || sd->sc.option&OPTION_XMAS
 			  || sd->sc.option&OPTION_SUMMER
 			  || sd->sc.option&OPTION_HANBOK
-			  || sd->sc.option&OPTION_OKTOBERFEST))
+			  || sd->sc.option&OPTION_OKTOBERFEST
```
Comment (Smokexyz) [inline src/map/status.c]: Spaces between binary bitwise operators.

File: src/map/status.c [hunk 6]
```diff
@@ -6883,13 +6885,16 @@ void status_set_viewdata(struct block_list *bl, int class_)
...
+			  || sd->sc.option&OPTION_SUMMER2))
```
Comment (Smokexyz) [inline src/map/status.c]: This could be compared as `!= 0` at the end?

File: src/map/clif.c (hunk 3 above)
Comment (Jedzkie) [inline src/map/clif.c]: Done.

File: src/map/status.c (hunk 5 above)
Comment (Jedzkie) [inline src/map/status.c]: Done.

File: src/map/status.c (hunk 6 above)
Comment (Jedzkie) [inline src/map/status.c]: Yep. Done.

File: src/map/clif.c [hunk 7]
```diff
@@ -3194,31 +3194,34 @@ void clif_changelook(struct block_list *bl,int type,int val)
...
+				if ((sd->sc.option & OPTION_OKTOBERFEST) != 0 && battle_config.oktoberfest_ignorepalette == true)
 					vd->cloth_color = 0;
-				if (sd->sc.option&OPTION_OKTOBERFEST /* TODO: config? */)
+				if ((sd->sc.option & OPTION_SUMMER2) != 0 && battle_config.summer2_ignorepalette == true)
 					vd->cloth_color = 0;
 				if (vd->body_style && (
-					sd->sc.option&OPTION_WEDDING || sd->sc.option&OPTION_XMAS ||
-					sd->sc.option&OPTION_SUMMER || sd->sc.option&OPTION_HANBOK ||
-					sd->sc.option&OPTION_OKTOBERFEST))
+					sd->sc.option & OPTION_WEDDING || sd->sc.option & OPTION_XMAS ||
```
Comment (MishimaHaruna) [inline src/map/clif.c]: This should be:

```
if (vd->body_style != 0 && (sd->sc.option & OPTION_COSTUME) != 0)
	vd->body_style = 0;
```

File: src/map/status.c [hunk 8]
```diff
@@ -6873,23 +6875,26 @@ void status_set_viewdata(struct block_list *bl, int class_)
...
+				if ((sd->sc.option & OPTION_SUMMER2) != 0 && battle_config.summer2_ignorepalette == true)
 					sd->vd.cloth_color = 0;
 			}
 			if (sd->vd.body_style
-			 && (sd->sc.option&OPTION_WEDDING
-			  || sd->sc.option&OPTION_XMAS
-			  || sd->sc.option&OPTION_SUMMER
-			  || sd->sc.option&OPTION_HANBOK
-			  || sd->sc.option&OPTION_OKTOBERFEST))
+			 && (sd->sc.option & OPTION_WEDDING
```
Comment (MishimaHaruna) [inline src/map/status.c]:
```
if (sd->vd.body_style != 0 && (sd->sc.option & OPTION_COSTUME) != 0)
	sd->vd.body_style = 0;
```

Comment (Jedzkie) [inline src/map/status.c]: done.

File: src/map/clif.c (hunk 7 above)
Comment (Jedzkie) [inline src/map/clif.c]: done.

**Comment by Jedzkie** [comment]
@MishimaHaruna any news on this?

**Comment by MishimaHaruna** [comment]
Merged, sorry for the delay!

**Pull request merged**